### ISSUE TITLE
Test viewlet: accessibility

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#114653]
- [x] `enter` and `space` when on test file should run the test 
- [x] aria label of test item should contain test status (successful, failed)
- [x] aria of view should contain total number of passed tests / total number of tests
- [x] There shuold be a keyboard way to navigate to test and move focus to editor (not sure how)
- [x] do an aria announcment once all tests finish running and announce how many passed how many failed
- [x] clicking on an item should focus that item (not its parent)
- [x] Typing characters while focus is on test items should type in the filter box. I do not think we need tree native filtering and filtering via the input box. I think it is confusing
- [x] zone widget which shows the expected vs actual should have a better aria label so when focus is on it we announce what is expected and what is actual value
- [x] zone widget should close on `esc` when focus is on it

Once we tackle this items I can share the test viewlet with our accessibility community so you might recieve additional feedback

**Comment by connor4312** [comment]
- total passed/failed tests now has visible text
- used alt+enter to focus in the editor

**Comment by connor4312** [comment]
> zone widget which shows the expected vs actual should have a better aria label so when focus is on it we announce what is expected and what is actual value

Where are you thinking this should be added? We have the peek view title -- which right shows the test `message` given by the test provider. Are you thinking something additional/instead of that?

**Comment by isidorn** [comment]
If the test `message` has enough information then we are good. But I just tried it and it does not read out what is expected and what is actual.
Since I think you are using an editor for each side, you should be able to set an `ariaLabel` of the editor. That ariaLabel for both the left and right side should contain the info what is expected and what is actual. That way whenever the focus is in those two small editors the right thing will be read out.
Let me know if this makes sense to you. Thanks!

**Comment by isidorn** [comment]
Also very cool that you tackled most of the things I mentioned in my inital comment already!

**Comment by connor4312** [comment]
I have tests run on "enter" but not "space", otherwise there's not a way to focus a test with the keyboard without running it.

**Comment by isidorn** [comment]
Thanks for improving this!
